plugins/DTI/systems/SpikeReportSystem.cpp

```diff
@@ -0,0 +1,214 @@
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "SpikeReportSystem.h"
+
+#include <brayns/engine/common/MathTypesOsprayTraits.h>
+#include <brayns/engine/components/Geometries.h>
+#include <brayns/engine/components/GeometryViews.h>
+#include <brayns/engine/components/SimulationInfo.h>
+#include <brayns/engine/geometry/types/Capsule.h>
+
+#include <api/StreamlineColorGenerator.h>
+#include <components/SpikeReportData.h>
+#include <components/StreamlineColors.h>
+
+namespace
+{
+/**
+ * @brief Applies default Streamline colors to the geometry view
+ */
+class DefaultColorPainter
+{
+public:
+    static void paint(brayns::Components &components)
+    {
+        auto &geometries = components.get<brayns::Geometries>();
+        auto &views = components.get<brayns::GeometryViews>();
+        auto &colors = components.get<dti::StreamlineColors>();
+
+        for (size_t i = 0; i < geometries.elements.size(); ++i)
+        {
+            auto capsules = geometries.elements[i].as<brayns::Capsule>();
+
+            auto &color = colors.elements[i];
+            color = dti::StreamlineColorGenerator::generate(*capsules);
+
+            auto &view = views.elements[i];
+            view.setColorPerPrimitive(ospray::cpp::SharedData(color));
+        }
+        views.modified = true;
+    }
+};
+
+/**
+ * @brief Applies a spike highlight to the streamline's geometry view
+ */
+class SimulationColorPainter
+{
+public:
+    static void paint(brayns::Components &components, const std::vector<std::vector<float>> &streamlineValues)
+    {
+        auto &views = components.get<brayns::GeometryViews>();
+        auto &geometries = components.get<brayns::Geometries>();
+        auto &colors = components.get<dti::StreamlineColors>();
+
+        for (size_t i = 0; i < streamlineValues.size(); ++i)
+        {
+            auto &streamlineData = streamlineValues[i];
+            if (streamlineData.empty())
+            {
+                continue;
+            }
+
+            auto &geometry = geometries.elements[i];
+            auto &color = colors.elements[i];
+            auto &view = views.elements[i];
+
+            auto primitiveCount = geometry.numPrimitives();
+            auto streamlineLength = primitiveCount - 1;
+
+            for (auto value : streamlineData)
+            {
+                auto index = static_cast<size_t>(value * streamlineLength);
+                index = std::clamp(index, 0ul, streamlineLength);
+                color[index] = brayns::Vector4f(1.f);
+                view.setColorPerPrimitive(ospray::cpp::SharedData(color));
+            }
+        }
+        views.modified = true;
+    }
+};
+
+/**
+ * @brief Check the status of the simulation (enabled, disabled, restore defaults, etc.)
+ */
+class SimulationChecks
+{
+public:
+    static bool enabled(const brayns::SimulationInfo &info, dti::SpikeReportData &data, brayns::Components &components)
+    {
+        if (!info.enabled)
```

Comment:
No need to change that but we could use guard clause instead of nested conditions.

Something like this:

if info.enabled: return true
if data.lastEnabledFlag: paint + reset
return false

Reply:
Actually I did it like that on the radiireportsystem, and I prefer it as well that way. Addressed.